File: frontend/src/questionnaires/dto/QuestionResponseEditorDto.ts
import {QuestionType} from "./QuestionType.ts";
import {AnswerResponseEditorDto} from "./AnswerResponseEditorDto.ts";

export interface QuestionResponseEditorDto {
  id: number;
  text: string;
  type: QuestionType;
  order: number;
  points: number;
  answers: AnswerResponseEditorDto[];
}

File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/QuestionResponseEditorDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import net.dnadas.training_portal.model.group.project.questionnaire.QuestionType;

import java.util.List;

public record QuestionResponseEditorDto(Long id, String text, QuestionType type, Integer points,
                                        Integer order,
                                        List<AnswerResponseEditorDto> answers) {
}
